radicle-cli/src/commands/ls.rs

```diff
@@ -48,18 +48,19 @@ pub fn run(_options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
     let storage = &profile.storage;
     let mut table = term::Table::default();
 
-    storage.repositories()?.into_iter().for_each(|id| {
-        let Ok(repo) = storage.repository(id) else { return };
-        let Ok((_, head)) = repo.head() else { return };
-        let Ok(proj) = repo.project_of(profile.id()) else { return };
+    for id in storage.repositories()?.into_iter() {
+        let Ok(repo) = storage.repository(id) else { break };
+        let Ok((_, head)) = repo.head() else { break };
+        let Ok(proj) = repo.project() else { break};
+
         let head = term::format::oid(head);
         table.push([
             term::format::bold(proj.name().to_owned()),
             term::format::tertiary(id.urn()),
             term::format::secondary(head),
             term::format::italic(proj.description().to_owned()),
         ]);
-    });
+    }
```

Comment:
Mh why we need to change the way that we do the for loop here? am I missing somethings?

Reply:
Returning errors out of the parent function is easier, and the documentation recommends 'for .. in .. {}' loops over this.

Reply:
Not needed but easier to read.

[Its unneeded but docum](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each)

"It’s generally more idiomatic to use a for loop, but for_each may be more legible when processing items at the end of longer iterator chains"